runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -623,24 +764,191 @@ public IClientRouter getRouter(String address) {
+                    logServerVersions();
+
+                    // If fast loading was requested, load maps before restarting.
+                    if (parameters.isUseFastLoader()
+                                && getObjectsView().getObjectCache().isEmpty()) {
+                        FastObjectLoader fastLoader = new FastObjectLoader(this)
+                                .setBatchReadSize(parameters.getBulkReadSize())
+                                .setTimeoutInMinutesForLoading(
+                                    (int) parameters.fastLoaderTimeout.toMinutes());
+                        fastLoader.loadMaps();
+                    }
+                    log.info("fetchLayout: Successfully installed new layout from {} in epoch {}",
+                            server, newLayout.getEpoch());
+                    return;
+                } catch (RequestTimeoutException | NetworkException | WrongEpochException
+                            | WrongClusterException | TimeoutException | NoBootstrapException e) {
+                    // We had a problem connecting to this layout server, so we retry using the
+                    // next server in the list.
+                    log.info("fetchLayout: {} connecting to {}",
+                            e.getClass().getSimpleName(), server, e);
+                }
+            }
+            // Run the system down handler.
+            try {
+                systemDownHandler.run();
+            } catch (SystemUnavailableError e) {
+                layout.completeExceptionally(e);
+                log.error("invalidateLayout: Terminating due to system unavailable request", e);
+                return;
+            }
+            log.warn("fetchLayout: Couldn't connect to any layout servers, retrying in {}",
+                    parameters.connectionRetryRate);
+            // Sleep and retry
+            Sleep.sleepUninterruptibly(parameters.connectionRetryRate);
+        }
+
+        log.warn("fetchLayout: Terminated due to shutdown, cancelling future.");
+        layout.completeExceptionally(new RuntimeShutdownError());
+        throw new RuntimeShutdownError();
+    }
+
+    /** Get a layout from a layout server.
+     *
+     * @param server                The server to fetch a {@link Layout} from.
+     * @return                      The {@link Layout} retrieved from the server.
+     * @throws TimeoutException     If the server timed out servicing the request.
+     * @throws NetworkException     If there was a network issue communicating to the server.
+     * @throws WrongEpochException  If the remote server was in the wrong epoch.
+     */
+    private Layout getLayoutFromServer(@Nonnull String server)
+            throws TimeoutException {
+        log.info("getLayoutFromServer: Trying {}", server);
+        IClientRouter router = getRouter(server);
+        // Try to get a layout.
+        Layout layout =
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "layout" which hides the field declared at line 391. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)